flask/db.py

```diff
@@ -36,49 +35,70 @@
-            result = dict(product)
-            result["reviews"] = []
+            span.set_tag("totalProducts",len(products))
+            span.set_data("products",products)
+        
+        with sentry_sdk.start_span(op="get_products.reviews", description="db.query") as span:
+            for product in products:
+                reviews = connection.execute(
+                    "SELECT * FROM reviews WHERE productId = {}".format(product.id)
+                ).fetchall()
+                result = dict(product)
+                result["reviews"] = []
+
+                for review in reviews:
+                    result["reviews"].append(dict(review))
+                results.append(result)
+            span.set_data("reviews", results)
 
-            for review in reviews:
-                result["reviews"].append(dict(review))
-            results.append(result)
-        return json.dumps(results, default=str)
+        with sentry_sdk.start_span(op="serialization", description="json"):
+            result = json.dumps(results, default=str)
+        return result
     except Exception as err:
         raise(err)
 
 # 2 sql queries max, then sort in memory
 def get_products_join():
     results = []
     try:
-        connection = db.connect()
-        products = connection.execute(
-            "SELECT * FROM products"
-        ).fetchall()
-
-        reviews = connection.execute(
-            "SELECT reviews.id, products.id AS productid, reviews.rating, reviews.customerId, reviews.description, reviews.created FROM reviews INNER JOIN products ON reviews.productId = products.id"
-        ).fetchall()
-
-        for product in products:
-            result = dict(product)
-            result["reviews"] = []
-
-            for review in reviews:
-                productId=review[1]
-                if productId == product["id"]:
-                    result["reviews"].append(dict(review))
-            results.append(result)
+        with sentry_sdk.start_span(op="get_products_join", description="db.connect"):
+            connection = db.connect()
+        
+        with sentry_sdk.start_span(op="get_products_join", description="db.query") as span:
+            products = connection.execute(
+                "SELECT * FROM products"
+            ).fetchall()
+            span.set_tag("totalProducts",len(products))
+            span.set_data("products",products)
 
-        return json.dumps(results, default=str)
+        with sentry_sdk.start_span(op="get_products_join.reviews", description="db.query") as span:
+            reviews = connection.execute(
+                "SELECT reviews.id, products.id AS productid, reviews.rating, reviews.customerId, reviews.description, reviews.created FROM reviews INNER JOIN products ON reviews.productId = products.id"
+            ).fetchall()
+            span.set_data("reviews",reviews)
+
+        with sentry_sdk.start_span(op="get_products_join.format_results", description="function") as span:
```

Comment:
span for formatting the results (we had a span for this in Tracing demo)

and then `span.set_data` so you can see what that post-formatted data actually is (we didn't do that in Tracing demo)

Think of this as akin to using setExtra or setAdditional with an error.